### PR TITLE
Edits for the Overview section

### PR DESCRIPTION
**Pull request by dempseyatgithub**
This PR contains suggested edits for the Overview section. It is an editorial pass not meant to change content.

In many cases, I have added comments here in GitHub to the changed lines themselves.

### CODE REVIEW

File: Guide.docc/MigrationGuide.md [hunk 1]
```diff
@@ -14,27 +14,27 @@
...
-Regardless of where your project is in this process, this guide exists to
-provide concepts and practical help to ease the migration.
+Regardless of where your project is in this process, this guide provides
+concepts and practical help to ease the migration.
 
 Here you will find articles and code examples that will:
 
 - Explain the concepts used by Swift's data-race safety model.
-- Help you enable complete concurrency checking with Swift 5 projects.
+- Show how to enable complete concurrency checking for Swift 5 projects.
```
Comment (dempseyatgithub) [inline Guide.docc/MigrationGuide.md:37]: Previously, only one bullet point referenced 'you', this change makes this item have the same style as the rest.

File: Guide.docc/MigrationGuide.md [hunk 2]
```diff
@@ -14,27 +14,27 @@
...
 languages exposed to Swift, can all interoperate with
 modules that have been migrated to Swift 6.
 
-It is possible you have been incrementally adopting all concurrency features
+It is possible you have been incrementally adopting concurrency features
 as they were introduced.
 Or, you may have been waiting for Swift 6 to begin using them.
-Regardless of where your project is in this process, this guide exists to
-provide concepts and practical help to ease the migration.
+Regardless of where your project is in this process, this guide provides
```
Comment (dempseyatgithub) [inline Guide.docc/MigrationGuide.md:31]: 'exists to' doesn't provide additional info

File: Guide.docc/MigrationGuide.md [hunk 3]
```diff
@@ -14,27 +14,27 @@
...
-Adopting the Swift 6 language mode is entirely under your control,
+Adopting the Swift 6 language mode is entirely under your control
 on a per-target basis.
-Targets that build with previous modes, as well as code that uses other
+Targets that build with previous modes, as well as code in other
 languages exposed to Swift, can all interoperate with
 modules that have been migrated to Swift 6.
 
-It is possible you have been incrementally adopting all concurrency features
+It is possible you have been incrementally adopting concurrency features
```
Comment (dempseyatgithub) [inline Guide.docc/MigrationGuide.md:28]: I think the intent of the 'all' was to convey 'whether you've been all-in on adopting concurrency, or you haven't done anything at all.'

I think the 'all' is implied in 'incrementally adopting concurrency features as they were introduced' and reads better.

Plus without the 'all' I think it reads as more inclusive for folks who have adopted some concurrency features already.

File: Guide.docc/MigrationGuide.md [hunk 4]
```diff
@@ -14,27 +14,27 @@
 
 Swift's concurrency system, introduced in [Swift 5.5](https://www.swift.org/blog/swift-5.5-released/),
 makes asynchronous and parallel code easier to write and understand.
-With the Swift 6 language mode, the compiler can now also
-guarantee that concurrent programs are data-race-free.
-When this mode is enabled, the compiler safety checks that were
-optional become required.
+With the Swift 6 language mode, the compiler can now 
+guarantee that concurrent programs are free of data races.
```
Comment (dempseyatgithub) [inline Guide.docc/MigrationGuide.md:18]: Avoids the somewhat awkward 'data-race-free'.

File: Guide.docc/MigrationGuide.md [hunk 5]
```diff
@@ -14,27 +14,27 @@
...
+With this mode enabled, compiler safety checks that were
```
Comment (dempseyatgithub) [inline Guide.docc/MigrationGuide.md:19]: I just realized this suggested change has two sentences in a row beginning with 'With' which is probably not a great thing.

I think that just starting this sentence with "When enabled, compiler safety checks..." it is clear enough that Swift 6 language mode is being referenced from the previous sentence.

File: Guide.docc/MigrationGuide.md [hunk 6]
```diff
@@ -14,27 +14,27 @@
 
 Swift's concurrency system, introduced in [Swift 5.5](https://www.swift.org/blog/swift-5.5-released/),
 makes asynchronous and parallel code easier to write and understand.
-With the Swift 6 language mode, the compiler can now also
-guarantee that concurrent programs are data-race-free.
-When this mode is enabled, the compiler safety checks that were
-optional become required.
+With the Swift 6 language mode, the compiler can now 
+guarantee that concurrent programs are free of data races.
+When enabled, compiler safety checks that were
```
Comment (dempseyatgithub) [inline Guide.docc/MigrationGuide.md:19]: I think it is clear that Swift 6 language mode from the previous sentence is being referred to by 'When enabled'.

**Review by mattmassicotte — APPROVED** [review]
Great changes, thank you so much.

**Pull request merged**